### PR TITLE
feat: 🎸 support project configs within nx workspace v2 

### PR DESCRIPTION
**Pull request by evanfuture**
✅ Closes: #116

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x ] The commit message follows our guidelines: https://github.com/ngneat/transloco/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/utils/resolve-project-base-path.ts [hunk 1]
```diff
@@ -24,9 +25,19 @@ export function resolveProjectBasePath(projectName?: string): {
...
+          break;
+        }
+        if (typeof project === 'string') {
+          const pConfig = searchConfig(projectConfig, project);
+          if (pConfig) {
+            sourceRoot = pConfig.sourceRoot;
+            projectType = pConfig.projectType;
+            break;
+          }
+        }
```
[the remark this answers is not in the document]
Comment (shaharkazaz) [reply src/utils/resolve-project-base-path.ts:40]: The error message should also be updated in this case, when you have a searchFrom.

File: src/utils/resolve-project-base-path.ts [hunk 2]
```diff
@@ -5,9 +5,10 @@ import { ProjectType } from '../config';
 
 const angularConfig = ['angular.json', '.angular.json'];
 const workspaceConfig = ['workspace.json'];
+const projectConfig = ['project.json'];
 
-function searchConfig(searchPlaces: string[]) {
-  return cosmiconfigSync('', { searchPlaces }).search()?.config;
+function searchConfig(searchPlaces: string[], searchFrom: string = '') {
```
Comment (shaharkazaz) [inline src/utils/resolve-project-base-path.ts:10]: When providing a default value the type is inffered.
```suggestion
function searchConfig(searchPlaces: string[], searchFrom = '') {
```

**Review by shaharkazaz — COMMENTED** [review]
Thanks for the PR!
Please see my comments and handle the error in case the project config is missing.
Also, after the fixes please force push this as a feature 🙂

**Comment by evanfuture** [comment]
Hey, sorry to leave this off for so long, work's been a lot lately.   In handling the errors, I ended up moving things around a bit.  I hope it's still easy to follow and you don't mind my removing the for loop in exchange for being more explicit about what's being returned when.

**Comment by JelleBruisten** [comment]
This pull request has been awaiting review for a while, can someone review this?

File: src/utils/validators.utils.ts [hunk 3]
```diff
@@ -4,6 +4,10 @@ export function isObject(value: any) {
   return value && typeof value === 'object' && !Array.isArray(value);
 }
 
+export function isString(value: any) {
+  return value && typeof value === 'string' && !Array.isArray(value);
+}
```
Comment (shaharkazaz) [inline src/utils/validators.utils.ts:9]:
```suggestion
export function isString(value: any): value is string {
  return typeof value === 'string';
}
```

File: src/utils/resolve-project-base-path.ts [hunk 4]
```diff
@@ -3,11 +3,27 @@ import { cosmiconfigSync } from 'cosmiconfig';
...
+const projectConfig = ['project.json'];
+const sourceRootDefault = 'src';
+
+function searchConfig(searchPlaces: string[], searchFrom = '') {
+  return cosmiconfigSync('', { searchPlaces }).search(searchFrom)?.config;
+}
 
-function searchConfig(searchPlaces: string[]) {
-  return cosmiconfigSync('', { searchPlaces }).search()?.config;
+function defaultWithError(places: string): {
```
Comment (shaharkazaz) [inline src/utils/resolve-project-base-path.ts:17]: I think it's best if this function only logs the error and the config is explicitly returned from the function.

File: src/utils/resolve-project-base-path.ts [hunk 5]
```diff
@@ -3,11 +3,27 @@ import { cosmiconfigSync } from 'cosmiconfig';
...
+  projectBasePath: string;
+  projectType: ProjectType;
+} {
+  console.log(
+    chalk.black.bgRed(
+      `Unable to load workspace config from ${places}. Defaulting source root to '${sourceRootDefault}'`
+    )
+  );
+  return { projectBasePath: sourceRootDefault, projectType: undefined };
```
Comment (shaharkazaz) [inline src/utils/resolve-project-base-path.ts:26]: Make `projectType` optional

File: src/utils/resolve-project-base-path.ts [hunk 6]
```diff
@@ -16,31 +32,31 @@ export function resolveProjectBasePath(projectName?: string): {
...
-      projectName = projectName || config.defaultProject;
-      const project = config.projects?.[projectName];
-      if (project) {
-        sourceRoot = project.sourceRoot;
-        projectType = project.projectType;
-        break;
-      }
-    }
+  const config = workspace || angular;
+  if (!config) {
```
Comment (shaharkazaz) [inline src/utils/resolve-project-base-path.ts:36]:
```suggestion
  if (!workspace || angular) {
```

File: src/utils/resolve-project-base-path.ts [hunk 7]
```diff
@@ -16,31 +32,31 @@ export function resolveProjectBasePath(projectName?: string): {
...
+    if (isString(project)) {
+      const pConfig = searchConfig(projectConfig, project);
+      if (pConfig) {
+        return {
+          projectBasePath: pConfig.sourceRoot,
+          projectType: pConfig.projectType,
+        };
+      }
+    }
   }
```
Comment (shaharkazaz) [inline src/utils/resolve-project-base-path.ts:59]:
```suggestion
  const config = isString(project) ? searchConfig(projectConfig, project) : project;
  if (project) {  
    return {
      projectBasePath: config.sourceRoot,
      projectType: config.projectType,
    };
  }
```

**Review by shaharkazaz — COMMENTED** [review]
Finally found the time 🙂 
Please update the `resolveProjectBasePath.spec` as well.

**Comment by evanfuture** [comment]
@shaharkazaz if you're already close to this, maybe you can take it over?  I'm having a tough time finding free time, and the other morning I wasn't able to get the tests working.

**Comment by shaharkazaz** [comment]
@evanfuture I see, I'll take it from here.
I appreciate the time and will, hope to see you contributing here in the future 😊

**Pull request closed without merging**